src/utils/update/mapper.utils.ts

```diff
@@ -121,3 +121,13 @@ export const lowerCaseAllWordsExceptFirstLetters = (country: string | undefined)
   }
   );
 };
+
+export const hasPublicRegisterDetails = (cp: CompanyProfile): boolean => {
+  if (cp.isOnRegisterInCountryFormedIn !== undefined){
+    return cp.isOnRegisterInCountryFormedIn;
+  } else if (cp.foreignCompanyDetails?.originatingRegistry?.name && cp.foreignCompanyDetails?.registrationNumber) {
```

Comment:
Minor but as both fields are mandatory on registration if yes selected, you could probably just check for one of them.